defhingejlib.py

```diff
@@ -0,0 +1,275 @@
+              str(elem.nodes[1].int_label))
+        return {'NODE2_NOTFOUND'}
+
+    # nodes' objects
+    n1OBJ = bpy.data.objects[n1]
+    n2OBJ = bpy.data.objects[n2]
+
+    # creation of line representing the dist
+    defhingeobj_id = 'dist_' + str(elem.int_label)
+    defhingecv_id = defhingeobj_id + '_cvdata'
+
+    # check if the object is already present. If it is, remove it.
+    if defhingeobj_id in bpy.data.objects.keys():
+        bpy.data.objects.remove(bpy.data.objects[defhingeobj_id])
+
+    # check if the curve is already present. If it is, remove it.
+    if defhingecv_id in bpy.data.curves.keys():
+        bpy.data.curves.remove(bpy.data.curves[defhingecv_id])
+
+    # get offsets
+    f1 = elem.offsets[0].value
+    f2 = elem.offsets[1].value
+    q1 = elem.rotoffsets[0].value
+    q2 = elem.rotoffsets[1].value
+
+    # assign coordinates of knots in global frame
+    R1 = n1OBJ.rotation_quaternion.to_matrix()
+    R2 = n2OBJ.rotation_quaternion.to_matrix()
+    p1 = n1OBJ.location + R1 * Vector((f1[0], f1[1], f1[2]))
+    p2 = n2OBJ.location + R2 * Vector((f2[0], f2[1], f2[2]))
+
+    bpy.ops.object.empty_add(type='PLAIN_AXES')
+    defhingeOBJ = bpy.context.scene.objects.active
+    defhingeOBJ.rotation_mode = 'QUATERNION'
+    defhingeOBJ.rotation_quaternion = n1OBJ.rotation_quaternion * Quaternion(( q1[0], q1[1], q1[2], q1[3] ))
+
+    # Set parent to n1OBJ
+    bpy.ops.object.select_all(action='DESELECT')
+    defhingeOBJ.select = True
+    n1OBJ.select = True
+    bpy.context.scene.objects.active = n1OBJ
+    bpy.ops.object.parent_set(type = 'OBJECT', keep_transform = False)
+    defhingeOBJ.location = p1
+
+    elem.blender_object = elem.name
+    defhingeOBJ.name = elem.name
+    ude = bpy.context.scene.mbdyn.elems_to_update.add()
+    ude.dkey = elem.name
+    ude.name = elem.name
+
+    q1h_tilde = Quaternion((q1))
+    rotation1 = n1OBJ.matrix_world.to_quaternion()
+    q2h_tilde = Quaternion((q1))
+    rotation2 = n2OBJ.matrix_world.to_quaternion()
+
+    q1h = q1h_tilde*rotation1
+    q2h = q2h_tilde*rotation2
+    qrel = q2h*q1h.conjugated()
+    n, theta = qrel.to_axis_angle()
+    phi = n*theta
+
+    length = 1
+    radius = 0.3 * length
+    print(length, radius)
+    turns = 7
+    axes = ['X', 'Y', 'Z']
+    for ii in range(3):
+        bpy.ops.mesh.curveaceous_galore(ProfileType='Helix', helixHeight=length,\
+                                        helixWidth=radius, helixEnd = 360* turns, helixPoints = 1000)
+        defhingeChild = bpy.context.scene.objects.active
+        bpy.ops.object.origin_set(type = 'ORIGIN_CENTER_OF_MASS')
+        track_axes = axes[:]
+        track_axes.remove(track_axes[ii])
+        defhingeChild.location = p1
+        defhingeChild.rotation_mode = 'QUATERNION'
+        axis_direction = Vector((0, 0, 0))
+        axis_direction[ii] = 1
+        defhingeChild.rotation_quaternion = Vector((0, 0, 1)).rotation_difference(axis_direction)
```

Comment:
Same thing as above: you are using the correct local components of the deformation vector, but you are applying them to the springs aligned along the global X, Y, Z axes.